[2.12.0] Erreur 500 - api/meta/datasets?orderby=dataset_name [POST]

Bonjour à tous,
**Version**
GN=2.12.0

**Description du bug**
Lorsque je me connecte à l'application avec certains utilisateurs, l'application ne parvient pas à récupérer les jeux de données. 
J'obtiens une Erreur 500 sur la requête POST api `geonature/api/meta/datasets?orderby=dataset_name`.

![image](https://user-images.githubusercontent.com/96542451/229725946-fe0f4c62-0af7-4f8b-9c76-9749927e1ebf.png)

Modules concernés : 
- Occtax
- Occhab
- Import
- Synthèse
- Validation


**Comportement attendue**
Récupérer la liste des JDDs disponibles pour l’utilisateur.

**Comment reproduire**
Il semble que les utilisateurs posant problème sont liés à des groupes que j'ai défini dont
- le profil est `Référent` (code 3)
- Avec/Sans `permissions de Sensibilité` sur le module GeoNature

Pour autant, je ne vois pas en quoi se profil serait plus bloquant qu'un autre. 

**Logs**
``` py
[2023-04-04 09:43:19 +0200] [2504] [ERROR] Exception on /meta/datasets [POST]
Traceback (most recent call last):
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/geonatureadmin/geonature/backend/geonature/core/gn_permissions/decorators.py", line 18, in decorated_view
    return view_func(*args, **kwargs)
  File "/home/geonatureadmin/geonature/backend/geonature/core/gn_meta/routes.py", line 155, in get_datasets
    return dataset_schema.jsonify(query.all(), many=True)
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/flask_marshmallow/schema.py", line 39, in jsonify
    data = self.dump(obj, many=many)
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/marshmallow/schema.py", line 557, in dump
    result = self._serialize(processed_obj, many=many)
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/marshmallow/schema.py", line 519, in _serialize
    return [
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/marshmallow/schema.py", line 520, in <listcomp>
    self._serialize(d, many=False)
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/marshmallow/schema.py", line 525, in _serialize
    value = field_obj.serialize(attr_name, obj, accessor=self.get_attribute)
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/marshmallow/fields.py", line 342, in serialize
    return self._serialize(value, attr, obj, **kwargs)
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/marshmallow/fields.py", line 1911, in _serialize
    return self._serialize_method(obj)
  File "/home/geonatureadmin/geonature/backend/geonature/utils/schema.py", line 22, in get_cruved
    return {action: obj.has_instance_permission(scope) for action, scope in scopes.items()}
  File "/home/geonatureadmin/geonature/backend/geonature/utils/schema.py", line 22, in <dictcomp>
    return {action: obj.has_instance_permission(scope) for action, scope in scopes.items()}
  File "/home/geonatureadmin/geonature/backend/geonature/core/gn_meta/models.py", line 529, in has_instance_permission
    if g.current_user == self.digitizer or g.current_user in self.user_actors:
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/sqlalchemy/orm/attributes.py", line 294, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/sqlalchemy/orm/attributes.py", line 728, in get
    value = callable_(state, passive)
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/sqlalchemy/orm/strategies.py", line 969, in __call__
    return strategy._load_for_state(state, passive)
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/sqlalchemy/orm/strategies.py", line 710, in _load_for_state
    self._invoke_raise_load(state, passive, "raise")
  File "/home/geonatureadmin/geonature/backend/venv/lib/python3.8/site-packages/sqlalchemy/orm/strategies.py", line 671, in _invoke_raise_load
    raise sa_exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: 'TDatasets.digitizer' is not available due to lazy='raise'
```
Il semble que les logs orientent l'origine du problème vers la lecture de l'objet `scopes` après la récupération du CRUVED.

--- a/backend/geonature/core/gn_meta/models.py
+++ b/backend/geonature/core/gn_meta/models.py
@@ -526,7 +526,7 @@ class TDatasets(db.Model):
         if scope == 0:
             return False
         elif scope in (1, 2):
-            if g.current_user == self.digitizer or g.current_user in self.user_actors:
+            if g.current_user.id_role == self.id_digitizer or g.current_user in self.user_actors:
                 return True
             if scope == 2 and g.current_user.organisme in self.organism_actors:
                 return True
@@ -831,7 +831,7 @@ class TAcquisitionFramework(db.Model):
         if scope == 0:
             return False
         elif scope in (1, 2):
-            if g.current_user == self.creator or g.current_user in self.user_actors:
+            if g.current_user.id_role == self.id_digitizer or g.current_user in self.user_actors:
                 return True
             if scope == 2 and g.current_user.organisme in self.organism_actors:
                 return True

--- a/backend/geonature/core/gn_meta/routes.py
+++ b/backend/geonature/core/gn_meta/routes.py
@@ -128,6 +128,10 @@ def get_datasets():
             joinedload("role"),
             joinedload("organism"),
         ),
+        # next relationships are joined for permission checks purpose:
+        joinedload("acquisition_framework").options(
+            joinedload("cor_af_actor"),
+        ),
     )
     only = [
         "+cruved",

--- a/backend/geonature/core/gn_meta/routes.py
+++ b/backend/geonature/core/gn_meta/routes.py
@@ -146,7 +146,7 @@ def get_datasets():
         only.append("+synthese_records_count")
 
     if "modules" in fields:
-        query = query.options(joinedload("modules"), joinedload("digitizer"))
+        query = query.options(joinedload("modules"))
         only.append("modules")
 
     dataset_schema = DatasetSchema(only=only)

